docs/studios/unity/unity-sdk/RGAction.mdx

```diff
@@ -1,116 +1,153 @@
 ---
-sidebar_label: 'RGAction'
+sidebar_label: 'Defining Bot Actions'
 ---
 
-# RGAction
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
 
-Represents an action that a bot can instruct the game object to take.
-This `RGAction` class should be inherited in a sub class, and developers
-will need to implement the `StartAction` and `GetActionName` function.
+# Defining Bot Actions
 
-In most RGAction implementations, it is necessary to find GameObjects within the scene based 
-on the game state. See the [`RGState`](./RGState) class for defining the game state and the 
-[`RGFindUtils`](./RGFindUtils) class for helper functions to locate GameObjects within your scene.
+This section refers to an "action" as any instruction that a bot can issue to its GameObject.
+This can be anything from movement to casting abilities to navigating UI components.
 
-:::info
-
-This component is likely to change soon to be more developer-friendly, by changing the
-Dictionary type used to a generic serializable class that you can define.
-
-:::
-
-## Properties
-
-None
+It is common for actions to interact with entities in the [game state](./RGState) and their corresponding GameObjects.
+The [`RGFindUtils`](./RGFindUtils) class contains utility functions for finding such GameObjects within the scene and is
+useful for these types of actions.
 
-## Class Methods
+## Using the `RGAction` Attribute
 
-### `public abstract string GetActionName()`
+Defining an action is as simple as adding the `RGAction` attribute to a method.
+This allows the Regression Games SDK to recognize this method as the entrypoint into a callable action from your [bot code](./creating-bots/configuration).
 
-The name of this action, which is used by the bot to request this specific action.
+:::info
+The `RGAction` attribute must be used within a component, a class that inherits from `MonoBehavior`, or a class whose parent inherits from `MonoBehavior`.
+:::
 
-#### Arguments
-None
+By default, the action's name matches the method's, but this can be overridden by passing a different name to `RGAction`.
+Any parameters that the method accepts will be valid arguments in your bot code. Their names and data types must match.
```

Comment:
Might want to clarify what "types must match" here - I believe this means that the arguments being passed from the bot code must match the primitive or serialized JSON types of the arguments?

Reply:
I would agree we need clarity here.  As I understand it.  The arguments from your bot, must match the argument names and types in the C# code.

Reply:
Correct - updated for clarity